R/concRespPlot2.R

```diff
@@ -0,0 +1,68 @@
+#' Concentration Response Plot - New
+#'
+#' This function takes in `concRespCore` or `tcplhit2_core` output and
+#' generates a basic plot of the observed concentration response data with the
+#' winning model curve. The function returns a ggplot object that is capable of
+#' taking additional layers of ggplot add-ons.
+#'
+#' @param row Output from `concRespCore` or `tcplhit2_core`, containing information
+#' about the winning curve fit of a compound.
+#' @param log_conc Logical argument. If `TRUE`, convert the concentration (x-axis)
+#' into log-10 scale. Defaults to `FALSE`.
+#'
+#' @return A ggplot object, a scatter plot of the concentration response data
+#' overlaid with the winning model curve.
+#'
+#' @export
+#'
+
+concRespPlot2 <- function(row, log_conc = FALSE) {
+
+  # get the winning model curve
+  fit_method <- row[,"fit_method"]
+
+  #reformat conc and resp as vectors
+  conc <- as.numeric(str_split(row[1,"conc"],"\\|")[[1]])
+  logc <- log10(conc)
```

Comment:
* You should only calculate the logc and logc_plot if log_conc is set to TRUE.
* Also, if 0 is in the concentration series and log_conc is set to TRUE we may want to throw an error OR provide a pseudo 0 with 1e-10 or some small value and throw a warning message indicating the issue as well as the fix.